[RM Ch 8.2.4] Multi-cloud Security (#2596)
cntt-n/CNTT: doc/ref_model/chapters/chapter08.md

The Cloud Services and the Cloud Resources Brokers provide value-added services in addition to the fundamental capabilities like service and resource discovery.  These Brokers are critical for a multi-cloud environment to function and utilise cloud specific plugins to perform the necessary activities. These Brokers can, for example, provision and manage environments with resources and services for Machine Learning (ML) services, Augmented/Virtual Reality, or specific industries.

<a name="8.2.4"></a>
### 8.2.4 Aspects of Multi-Cloud Security

> As cloud infrastructures become a key element in the telecommunication operator ecosystem, particularly with the 5G rollout becoming a critical business necessity, cloud-focused threats and associated adversarial behaviours, methods, tools, and strategies that cyber threat actors use to plan and execute are part of the attack surface landscape.

Reviewer: Suggestion to break this one-sentence paragraph:
```suggestion
Cloud infrastructures, emerging as a key element in the telco operator ecosystem, are part of the attack surface landscape. This is particularly worrying with the 5G rollout becoming a critical business necessity. It is important to be vigilant of Cloud-focused threats and associated adversarial behaviours, methods, tools, and strategies that cyber threat actors use.
```

> In an ecosystem comprised of different network domains, products and business partnerships, the responsibility for managing these different cloud environments necessary to support 5G use cases falls to different organisations, creating a new level of complexities and a new range of security risks. In such an environment, there are additional security principles to be considered.  These principles, see Table 2 below, drawn from the collaboration with GSMA FASG.

Reviewer: ```suggestion
In the multi-cloud ecosystem comprised of different network domains, products, and business partnerships, the responsibility for managing these different cloud environments necessary to support 5G use cases falls to different enterprises, creating new levels of complexities and a new range of security risks. In such an environment, there are additional security principles to be considered.  These principles, see Table 2 below, are drawn from the collaboration with GSMA FASG.
```

  
| Multi-cloud Security Principle | Description |
| ------------------------------ | ----------- |
| Policy synchronization | Consistency in applying the right security policies across environments, services, interfaces, and configured resources |

> | Visibility | A common data model approach to capture events and behaviours across all the key compute, storage, network, and applications resources, environments, virtualised platforms, containers and interfaces |

Reviewer: Across different enterprises, a common data model for "capture" may not be feasible at least for a while until standards are formulated and their compliance becomes standard practice.  Can we substitute "share" for "capture"?

Author: @pgoyal01, I agree it is aspirational. Good idea to use "share".  I am changing it that way.

> | Monitoring | Centralisation, correlation, and visualisation of security information across the different cloud environments to provide an end-to-end view and enable timely response to attacks |

Reviewer: We may need to look at the networking model where instead of raw information shared by EMS to NMS abstractions are shared.

The timely response to individual instances still falls to the affected enterprise.  The central correlation can point to an attack on multiple enterprises with the intent of disabling or comprising the whole.

Not sure how to capture this.

Author: @pgoyal01:I am not sure that this fits into this section. Can we have a separate discussion, perhaps in wiki on this?

| Automation | Automation of critical activities including cloud security posture management, continuous security assessments, compliance monitoring, detection of misconfigurations and identification and remediation of risks |
| Access Management | Organisation of a wide range of users including administrators, testers, DevOps, and developers and customers should be organised into security groups with privileges appropriate to different resources and environments |
| Security Operations Model | Augmentation of security services provided by cloud service providers with the vetted third-party and/or open-source tools and services, all incorporated into the established overall security operations model |

> 

Reviewer: I think we need to consider the autonomous nature of the participants in the multi-cloud. We can prescribe certain things and if not satisfied treat that party as "untrusted", etc. 

Many of the above requirements may need to be reviewed with that in mind.

Author: I have added a paragraph (at the end of the section) addressing this to some extent.

<p align="center"><b>Table 8-1a. Multi-Cloud Principles</b></p>

> If telecommunication operators decide to run some of their public telecom network functions in a multi-cloud environment, and specifically in public clouds, the industry will need a set of new standards and new security tools able to regulate the interactions between participating parties. For instance, it seems to be sensible to eliminate access to some restricted areas altogether rather than merely prohibit it. The ETSI standard TS 103 457 “Interface to offload sensitive functions to a trusted domain” provides extra security requirements for public clouds to offer telecommunication operators the option of running public telecom network functions in public clouds.

Reviewer: ```suggestion
For telco operators to run their network functions in a multi-cloud environment, and specifically, in public clouds, the industry will need a set of new standards and new security tools to manage and regulate the interactions between multi-cloud participating parties. For instance, it seems sensible to eliminate access to restricted areas rather than merely prohibit it. The ETSI standard TS 103 457 “Interface to offload sensitive functions to a trusted domain” provides extra security requirements for public clouds so as to enable telco operators the option of running network functions in public clouds.
```

> If telecommunication operators decide to run some of their public telecom network functions in a multi-cloud environment, and specifically in public clouds, the industry will need a set of new standards and new security tools able to regulate the interactions between participating parties. For instance, it seems to be sensible to eliminate access to some restricted areas altogether rather than merely prohibit it. The ETSI standard TS 103 457 “Interface to offload sensitive functions to a trusted domain” provides extra security requirements for public clouds to offer telecommunication operators the option of running public telecom network functions in public clouds.

Reviewer 2: 1st I find to be clear good practice and not just sensible. 2nd unless it will be referred to later, could be softer as an example. Or something similar. In:
```suggestion
If telecommunication operators decide to run some of their public telecom network functions in a multi-cloud environment, and specifically in public clouds, the industry will need a set of new standards and new security tools able to regulate the interactions between participating parties. For instance, a good practice is to eliminate access to some restricted areas altogether rather than merely prohibit it. Also for example, the ETSI standard TS 103 457 “Interface to offload sensitive functions to a trusted domain” provides extra security requirements for public clouds to offer telecommunication operators the option of running public telecom network functions in public clouds.
```

Author: @petorre, changed according to your suggestions, thanks!

The Cloud Services and the Cloud Resources Brokers provide value-added services in addition to the fundamental capabilities like service and resource discovery.  These Brokers are critical for a multi-cloud environment to function and utilise cloud specific plugins to perform the necessary activities. These Brokers can, for example, provision and manage environments with resources and services for Machine Learning (ML) services, Augmented/Virtual Reality, or specific industries.

<a name="8.2.4"></a>
### 8.2.4 Aspects of Multi-Cloud Security
Cloud infrastructures, emerging as a key element in the telco operator ecosystem, are part of the attack surface landscape. This is particularly worrying with the 5G rollout becoming a critical business necessity. It is important to be vigilant of Cloud-focused threats and associated adversarial behaviours, methods, tools, and strategies that cyber threat actors use.
 

> In the multi-cloud ecosystem comprised of different security postures and policies, network domains, products, and business partnerships, the responsibility for managing these different cloud environments necessary to support 5G use cases falls to different enterprises, creating new levels of complexities and a new range of security risks. In such an environment, there are additional security principles to be considered.  These principles, see Table 2 below, are drawn from the collaboration with GSMA FASG.

Reviewer: Table 2? 
```suggestion
In the multi-cloud ecosystem comprised of different security postures and policies, network domains, products, and business partnerships, the responsibility for managing these different cloud environments necessary to support 5G use cases falls to different enterprises, creating new levels of complexities and a new range of security risks. In such an environment, there are additional security principles to be considered.  These principles, see Table 8-1a below, are drawn from the collaboration with the GSMA FASG.
```

> In the multi-cloud ecosystem comprised of different security postures and policies, network domains, products, and business partnerships, the responsibility for managing these different cloud environments necessary to support 5G use cases falls to different enterprises, creating new levels of complexities and a new range of security risks. In such an environment, there are additional security principles to be considered.  These principles, see Table 2 below, are drawn from the collaboration with GSMA FASG.

Reviewer 3: Can you please resolve GSMA FASG?

  
| Multi-cloud Security Principle | Description |
| ------------------------------ | ----------- |
| Policy synchronization | Consistency in applying the right security policies across environments, services, interfaces, and configured resources |
| Visibility | A common data model approach to share events and behaviours across all the key compute, storage, network, and applications resources, environments, virtualised platforms, containers and interfaces |
| Monitoring | Centralisation, correlation, and visualisation of security information across the different cloud environments to provide an end-to-end view and enable timely response to attacks |
| Automation | Automation of critical activities including cloud security posture management, continuous security assessments, compliance monitoring, detection of misconfigurations and identification and remediation of risks |

> | Access Management | Organisation of a wide range of users including administrators, testers, DevOps, and developers and customers should be organised into security groups with privileges appropriate to different resources and environments |

Reviewer 4: @walterkozlowski "should be organised" can be removed, it is redundant with the beginning of the sentence "organisation of"

Author: Replaced by: "Wide range of users including administrators, testers, DevOps, and developers and customers should be organised into security groups with privileges appropriate to different resources and environments" following advice form @karinesevilla

| Security Operations Model | Augmentation of security services provided by cloud service providers with the vetted third-party and/or open-source tools and services, all incorporated into the established overall security operations model |

<p align="center"><b>Table 8-1a. Multi-Cloud Principles</b></p>

> For telco operators to run their network functions in a multi-cloud environment, and specifically, in public clouds, the industry will need a set of new standards and new security tools to manage and regulate the interactions between multi-cloud participating parties. For instance, it is a good practice to eliminate access to restricted areas rather than merely prohibit it. For instance, the ETSI standard TS 103 457 “Interface to offload sensitive functions to a trusted domain” provides extra security requirements for public clouds so as to enable telco operators the option of running network functions in public clouds.

Reviewer 3: ```suggestion
For telco operators to run their network functions in a multi-cloud environment, and specifically, in public clouds, the industry will need a set of new standards and new security tools to manage and regulate the interactions between multi-cloud participating parties. For instance, it is a good practice to eliminate access to restricted areas rather than merely prohibit it. For instance, the ETSI specification [TS 103 457](https://www.etsi.org/deliver/etsi_ts/103400_103499/103457/01.01.01_60/ts_103457v010101p.pdf) “Interface to offload sensitive functions to a trusted domain” provides extra security requirements for public clouds so as to enable telco operators the option of running network functions in public clouds.
```

> For telco operators to run their network functions in a multi-cloud environment, and specifically, in public clouds, the industry will need a set of new standards and new security tools to manage and regulate the interactions between multi-cloud participating parties. For instance, it is a good practice to eliminate access to restricted areas rather than merely prohibit it. For instance, the ETSI specification [TS 103 457](https://www.etsi.org/deliver/etsi_ts/103400_103499/103457/01.01.01_60/ts_103457v010101p.pdf) “Interface to offload sensitive functions to a trusted domain” provides extra security requirements for public clouds so as to enable telco operators the option of running network functions in public clouds.

Reviewer 4: @walterkozlowski The sense of the sentence "For instance, it is a good practice to eliminate access to restricted areas rather than merely prohibit it" is unclear for me. I think something is missing to understand what it stands for.

Author: @karinesevilla: I have changed to "For instance, it is a good practice to eliminate physical access to restricted areas rather than merely prohibit it".  Does it clarify it?

Reviewer 4: @walterkozlowski  Not really, because I don't see how 'eliminate the physical access' can be an example of  'new standards and new security tools'

Author: @karinesevilla, when standards like ETSI specification [TS 103 457] are applied defining the clear interface between more and less trusted domains, it will introduce elimination of access.  However, I am happy to remove this sentence (which was proposed by someone else) if it creates confusion.  It was just meant to be an example but it seems not to play its role properly.  I will remove this sentence and reshape a bit the next one (grammatically only).

Reviewer 4: @walterkozlowski It is good for me, thanks

> There is also another security aspect to consider, which is related to the autonomous nature of the participants in the multi-cloud. We can prescribe certain things and if not satisfied treat that party as "untrusted".  This problem has been attcked to some extent in TS 103 457. This standard introduces a concept of an LTD (Less Trusted Domain) and an MTD (More Trusted Domain) and specifies the TCDI (Trusted Cross-Domain Interface) to standardise secure interactions between them.  The standard defined the following elementary functions of TCDI:

Reviewer: ```suggestion
There is also another security aspect to consider, which is related to the autonomous nature of the participants in the multi-cloud. We can prescribe certain things and if not satisfied treat that party as "untrusted".  This problem has been attacked to some extent in TS 103 457. This standard introduces a concept of an LTD (Less Trusted Domain) and an MTD (More Trusted Domain) and specifies the TCDI (Trusted Cross-Domain Interface) to standardise secure interactions between them.  The standard defined the following elementary functions of TCDI:
```

Author: @pgoyal01, please check that your change is done

> There is also another security aspect to consider, which is related to the autonomous nature of the participants in the multi-cloud. We can prescribe certain things and if not satisfied treat that party as "untrusted".  This problem has been attacked to some extent in TS 103 457. This standard introduces a concept of an LTD (Less Trusted Domain) and an MTD (More Trusted Domain) and specifies the TCDI (Trusted Cross-Domain Interface) to standardise secure interactions between them.  The standard defined the following elementary functions of TCDI:

Reviewer 2: ```suggestion
There is also another security aspect to consider, which is related to the autonomous nature of the participants in the multi-cloud. We can prescribe certain things and if not satisfied treat that party as "untrusted".  This problem has been addressed to some extent in TS 103 457. This standard introduces a concept of an LTD (Less Trusted Domain) and an MTD (More Trusted Domain) and specifies the TCDI (Trusted Cross-Domain Interface) to standardise secure interactions between them.  The standard defined the following elementary functions of TCDI:
```

Connection and session management
Data and value management

> Transferring cryptography functionality

Reviewer 2: ```suggestion
Transferring cryptography functionality:
```

- Entropy request
- Encryption keys request
- Trusted timestamping
- Secure archive
- Secure storage
- Search capabilities

> As described in Sec. 1 (Scope) of this document, it “… specifies a high-level service-oriented interface, as an application layer with a set of mandatory functions, to access secured services provided by, and executed in a More Trusted Domain. The transport layer is out of scope and left to the architecture implementation “. The standard provides extra security for sensitive functions down to individual Virtual Machines or Containers. As such, it is recommended that the relevant components of reference models, reference architecture, reference implementations and reference compliance take notice of this standard and ensure their compatibility, wherever possible.

Reviewer: ```suggestion
As described in Sec. 1 (Scope) of the TS 103 45 document, it specifies "… a high-level service-oriented interface, as an application layer with a set of mandatory functions, to access secured services provided by, and executed in a More Trusted Domain. The transport layer is out of scope and left to the architecture implementation". The standard provides extra security features for sensitive functions down to individual Virtual Machines or Containers. As such, it is recommended that the relevant components of reference models, reference architecture, reference implementations and reference compliance take notice of this standard and ensure their compatibility, wherever possible.
```
